installation-bundle/src/Config/ParameterDumper.php

```diff
@@ -34,14 +35,18 @@ class ParameterDumper
 
     public function __construct(string $rootDir, Filesystem $filesystem = null)
     {
-        $this->configFile = $rootDir.'/config/parameters.yml';
+        $this->configFile = Path::join($rootDir, 'config/parameters.yml');
         $this->filesystem = $filesystem ?: new Filesystem();
 
-        // Fallback to the legacy config file (see #566)
-        if (file_exists($rootDir.'/app/config/parameters.yml') && !file_exists($rootDir.'/config/parameters.yml')) {
-            $this->configFile = $rootDir.'/app/config/parameters.yml';
-        } elseif (!file_exists($this->configFile)) {
-            return;
+        if (!$this->filesystem->exists($this->configFile)) {
```

Comment:
Why did you change the logic here?

Reply:
If I understand it correctly the `file_exists` for `$this->configFile` now only gets executed once instead of twice.

Reply:
Before the paths were built 4 times, now twice. The logic is the same but the fallback is not checked prematurely resulting in less `file_exist` checks for some cases:

| default exists | fallback exists | =>       | checks old code                       | checks new code                       |
|----------------|-----------------|----------|---------------------------------------|---------------------------------------|
| yes            | yes             | default  | 2x exist(default), 1x exist(fallback) | 1x exist(default)                     |
| yes            | no              | default  | 1x exist(default), 1x exist(fallback) | 1x exist(default)                     |
| no             | yes             | fallback | 1x exist(default), 1x exist(fallback) | 1x exist(default), 1x exist(fallback) |
| no             | no              | -        | 1x exist(default), 1x exist(fallback) | 1x exist(default), 1x exist(fallback) |

Reply:
Do we have a unit test to verify that the method still behaves the same?

Reply:
I added one.

* Here is the proof: 7fa64ae4dd400c5f7c4a2513613e7642b1c6a5e7 (I just replaced `file_exist()` with `$filesystem->exists()`.)
* Here is the new version again: a11de745851b55acbc69368aeabb9be5eb98bfe2.